src/usb_moded-config.c

```diff
@@ -334,12 +335,26 @@ int set_config_setting(const char *entry, const char *key, const char *value)
 {
   GKeyFile *settingsfile;
   gboolean test = FALSE;
-  int ret = 0; 
+  int ret = SET_CONFIG_ERROR;
   gchar *keyfile;
 
   settingsfile = g_key_file_new();
   test = g_key_file_load_from_file(settingsfile, FS_MOUNT_CONFIG_FILE, G_KEY_FILE_NONE, NULL);
-  if(!test)
```

Comment:
I have been trying to minimize the changes. Feel free to rearrange the code the way you like.